File: Java/BookstoreAPI/src/main/java/com/bookstore/endpoints/BookService.java
package com.bookstore.endpoints;

import com.bookstore.BookstoreRESTApp;
import com.bookstore.models.Book;
import io.swagger.annotations.Api;
import org.springframework.stereotype.Component;
import org.springframework.web.bind.annotation.RequestBody;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import java.util.List;

/**
 * Created by GOA on 7/9/16.
 */

@Component
@Path("/api/books")
@Api(value = "BookService", produces = "application/json")
public class BookService {

    @GET
    @Path("/")
    @Produces(MediaType.APPLICATION_JSON)
    public List<Book> getAllBooks() {
        return BookstoreRESTApp.getInstance().getBookRepository().findAll();
    }

    @GET
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public Book getBookById(@PathParam("id") String id) {
        return BookstoreRESTApp.getInstance().getBookRepository().findOne(id);
    }

    @POST
    @Path("/")
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    public String addBook(@RequestBody Book book) {
        BookstoreRESTApp.getInstance().getBookRepository().insert(book);
        return "OK";
    }

    @PUT
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    public String updateBook(@RequestBody Book book) {
        BookstoreRESTApp.getInstance().getBookRepository().save(book);
        return "OK";
    }

    @DELETE
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public String deleteBook(@PathParam("id") String id) {
        BookstoreRESTApp.getInstance().getBookRepository().delete(BookstoreRESTApp.getInstance().getBookRepository().findOne(id));
        return "OK";
    }
}

File: Node.js/bookstore/public/js/services/BookService.js
var bookService = angular.module("bookService", []);

bookService.factory("BookService", function($http) {
  var services = {};

  services.getAllBooks = function() {
    return $http.get('http://localhost:8081/api/books').then(function(response) {
        return response.data;
    });
  };

  services.getBookById = function(id) {
    return $http.get('http://localhost:8081/api/books/'+id).then(function(response) {
        return response.data;
    });
  }

  services.deleteBookById = function(id) {
    return $http.delete('http://localhost:8081/api/books/'+id).then(function(response) {
        return response.data;
    });
  }

  services.addBook = function(book) {
    return $http.post('http://localhost:8081/api/books/', book).then(function(response) {
        return response.data;
    })
  }

  services.updateBook = function(book) {
    return $http.put('http://localhost:8081/api/books/' + book._id, book).then(function(response) {
        return response.data;
    })
  }

  return services;
});
